File: backend/app/schemas/labels.py
from pydantic import BaseModel, Field


# Coarse label taxonomy — expand by just adding more strings.
DEFAULT_LABEL_OPTIONS = [
    "walking",
    "running",
    "rest",
    "sleep",
    "stress",
    "artifact",
    "custom",
]


class LabelIn(BaseModel):
    usubjid: str
    study_day_start: float
    study_day_end: float
    label: str = Field(pattern="^(walking|running|rest|sleep|stress|artifact|custom)$")
    custom_label: str | None = None
    notes: str | None = None


class Label(LabelIn):
    label_id: str
    user_email: str | None = None
    created_at: str  # ISO 8601


class LabelList(BaseModel):
    items: list[Label]


class LabelOptions(BaseModel):
    options: list[str] = Field(default_factory=lambda: list(DEFAULT_LABEL_OPTIONS))

File: frontend/src/api/labels.ts
export const LABEL_OPTIONS = [
  'walking',
  'running',
  'rest',
  'sleep',
  'stress',
  'artifact',
  'custom',
] as const

export type LabelKind = (typeof LABEL_OPTIONS)[number]

export const LABEL_COLOR: Record<LabelKind, string> = {
  walking: '#087A6A',
  running: '#D35C65',
  rest: '#7AAD4E',
  sleep: '#4F7EE0',
  stress: '#E0A94F',
  artifact: '#7a7a7a',
  custom: '#A25BC5',
}

export interface LabelIn {
  usubjid: string
  study_day_start: number
  study_day_end: number
  label: LabelKind
  custom_label?: string | null
  notes?: string | null
}

export interface Label extends LabelIn {
  label_id: string
  user_email: string | null
  created_at: string
}

export interface LabelList {
  items: Label[]
}
